scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,607 @@
+##############################################################################
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
```

Comment:
As discussed on the mailing list, GPL code cannot be added to Scipy. The easiest is if you just remove this license statement, then your code falls under the license of Scipy (BSD). Or if you prefer you can add an explicit BSD-compatible license.